### PR TITLE
deps: Update all backend dependencies

### PR DESCRIPTION
**Pull request by jkroepke**
Before looking into #41, I would like to start with the lastest dependencies.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ jobs:
         if: steps.check-for-backend.outputs.has-backend == 'true'
         uses: actions/setup-go@v3
```
Comment (mahendrapaipuri) [inline .github/workflows/ci.yml:57]: Could you bump all instances of `setup-go` to `v5`?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -13,6 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
```
Comment (mahendrapaipuri) [inline .github/workflows/release.yml:15]: Could you bump all instances of `checkout` to `v4`?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -13,6 +13,7 @@ jobs:
...
-      - uses: grafana/plugin-actions/build-plugin@release
+      - uses: grafana/plugin-actions/build-plugin@0315492b375faa7af0fa6c226ecb6c93c6fe9745
```
Comment (mahendrapaipuri) [inline .github/workflows/release.yml:16]: I know that `release` tag is old but how did you choose this commit? Is it the latest one? If so, please add a comment saying that this is the latest commit on this day.

**Review by mahendrapaipuri — CHANGES_REQUESTED** [review]
First of all, thanks a ton for all the work you made to put up those PRs. I will look into them as soon as I can. They all look great at the first glance. I have some suggestions so that we can discuss them on each PR.

Regarding the curren PR, I have very few minor comments. I think we need to update [Validate plugin](https://github.com/mahendrapaipuri/grafana-dashboard-reporter-app/blob/88d0c230b07ce19f896a6960935ca1a55e7cd959/.github/workflows/ci.yml#L114-L120) step as well as it uses the `1.21.x` Go as of now.

File: .github/workflows/release.yml (hunk 3 above)
Comment (jkroepke) [reply .github/workflows/release.yml:16]:
```suggestion
      - uses: grafana/plugin-actions/build-plugin@0315492b375faa7af0fa6c226ecb6c93c6fe9745 # commit of 2024-07-08
```

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jkroepke) [reply .github/workflows/ci.yml:57]:
```suggestion
        uses: actions/setup-go@v5
```

[the remark this answers is not in the document]
Comment (jkroepke) [reply .github/workflows/ci.yml:57]:
```suggestion
        uses: actions/setup-go@v5
```

Comment (jkroepke) [reply .github/workflows/ci.yml:57]:
```suggestion
        uses: actions/setup-go@v5
```

File: .github/workflows/release.yml (hunk 2 above)
Comment (jkroepke) [reply .github/workflows/release.yml:15]:
```suggestion
      - uses: actions/checkout@v4
```

**Comment by mahendrapaipuri** [comment]
Thanks for the changes @jkroepke 

> Regarding the curren PR, I have very few minor comments. I think we need to update [Validate plugin](https://github.com/mahendrapaipuri/grafana-dashboard-reporter-app/blob/88d0c230b07ce19f896a6960935ca1a55e7cd959/.github/workflows/ci.yml#L114-L120) step as well as it uses the 1.21.x Go as of now.

We are still missing this one. Could you please update it?

**Pull request merged**